File: JunsBlog/ClientApp/src/app/models/article/articleSearchPagingOption.ts
import { SortOrderEnum } from '../enums/sortOrderEnum';
import { SortByEnum } from '../enums/sortByEnum';
import { ArticleFilterEnum } from '../enums/articleFilterEnum';


export class ArticleSearchPagingOption {
    currentPage: number;
    pageSize: number;
    searchKey: string;
    sortBy: SortByEnum;
    sortOrder: SortOrderEnum;
    filter: ArticleFilterEnum;
    profilerId: string;

    constructor(currentPage: number = 1, pageSize: number = 10, searchKey: string = null, 
        sortBy: SortByEnum = SortByEnum.CreatedOn, sortOrder: SortOrderEnum = SortOrderEnum.descending,
         filter: ArticleFilterEnum = ArticleFilterEnum.All, profilerId: string = null){
            this.currentPage = currentPage;
            this.pageSize = pageSize;
            this.searchKey = searchKey;
            this.sortBy = sortBy;
            this.sortOrder = sortOrder;
            this.filter = filter;
            this.profilerId = profilerId;
    }
}

File: JunsBlog/Models/Articles/ArticleSearchPagingOption.cs
using JunsBlog.Models.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace JunsBlog.Models.Articles
{
    public class ArticleSearchPagingOption
    {
        public string SearchKey { get; set; }
        public SortOrderEnum SortOrder { get; set; }
        public SortByEnum SortBy { get; set; }
        public int CurrentPage { get; set; }
        public int PageSize { get; set; }
        public ArticleFilterEnum Filter { get; set; }

        public ArticleSearchPagingOption()
        {
            SearchKey = String.Empty;
            SortOrder = SortOrderEnum.Descending;
            SortBy = SortByEnum.UpdatedOn;
            CurrentPage = 1;
            PageSize = 10;
            Filter = ArticleFilterEnum.All;
        }
    }
}
